analytics/transform_test.go

```diff
@@ -315,6 +305,24 @@ func TestGetValue(t *testing.T) {
 	assert.NotNil(err)
 }
 
+func TestGetUnstructEventValue(t *testing.T) {
+	assert := assert.New(t)
+
+	// correct value unstruct field
+	unstructValue, err := fullEvent.GetUnstructEventValue("unstruct_event")
+	assert.Nil(err)
+	assert.Equal(unstructMap, unstructValue)
+}
+
+func TestGetContextValue(t *testing.T) {
+	assert := assert.New(t)
+
+	// correct value contexts
+	contextsValue, err := fullEvent.GetContextValue("contexts")
+	assert.Nil(err)
+	assert.Equal(contextsMap, contextsValue)
+}
+
```

Comment:
It'd be nice to see some additional tests to cover the more complete usage of these functions, such as the ones you describe in the PR:

```
GetContextValue(`derived_contexts`, `contexts_nl_basjes_yauaa_context_1`, `agentClass`)

GetUnstructEventValue(`unstruct_event`, `unstruct_event_com_snowplowanalytics_snowplow_add_to_cart_1`, `currency`, 0)
```